pkg/scheduler/scheduler.go

```diff
@@ -423,11 +420,15 @@ func (e entryOrdering) Less(i, j int) bool {
 	return a.Obj.CreationTimestamp.Before(&b.Obj.CreationTimestamp)
 }
 
-func (s *Scheduler) requeueAndUpdate(log logr.Logger, ctx context.Context, w *workload.Info, message string) {
-	added := s.queues.RequeueWorkload(ctx, w)
-	log.V(2).Info("Workload re-queued", "workload", klog.KObj(w.Obj), "queue", klog.KRef(w.Obj.Namespace, w.Obj.Spec.QueueName), "added", added)
-	err := workload.UpdateStatus(ctx, s.client, w.Obj, kueue.WorkloadAdmitted, corev1.ConditionFalse, "Pending", message)
-	if err != nil {
-		log.Error(err, "Could not update Workload status")
+func (s *Scheduler) requeueAndUpdate(log logr.Logger, ctx context.Context, e entry, message string) {
+	added := s.queues.RequeueWorkload(ctx, &e.Info, e.status != skipped)
+	log.V(2).Info("Workload re-queued", "workload", klog.KObj(e.Obj), "queue", klog.KRef(e.Obj.Namespace, e.Obj.Spec.QueueName), "added", added, "status", e.status)
+
+	if e.status != skipped {
```

Comment:
```suggestion
	if e.status == nominated {
```

So that we only do it if it didn't fit.

Reply:
if so we can't update the status and event for assumed  workload.

Reply:
maybe keep `e.status != skipped` .

Reply:
We don't do that. We just do the spec update in that case.

Reply:
We actually want to avoid the status update and event for `assumed` status. The only way we get here is when the spec update failed.

Reply:
wait, we should keep this as is, or introduce an explicit inadmissible entryStatus because entries with no status set (meaning they weren't even nominated) should be included here.

Reply:
also, we should remove the status update that happens here: https://github.com/kubernetes-sigs/kueue/blob/e9f4c4dc3e6758d031a3adabf4cbd80d29de3256/pkg/scheduler/scheduler.go#L124

Reply:
Removed

Reply:
To summarize, we get to `requeueAndUpdate` in the following statuses

1. status not set: workload didn't fit at all.
2. skipped: workload was nominated but then skipped because the cohort was used.
3. nominated: admit failed because of Spec update failure.
4. we never get here in assumed status

Reply:
3 only happens if we are trying to assume an already assumed workload or the clusterqueue was removed while we were scheduling.

we do get here with 4, but from the `admit` sub-routine.

I think we only should do the update + event in case 1.

Reply:
unit tests? :)

Reply:
right, admit is non-blocking and starts a go routine.

+1 to unit tests and only update status + event on the inadmissible case only

Reply:
Added unit tests to the changed part.